internal/cache/controller_obj_cache.go

```diff
@@ -2842,11 +2842,57 @@ func checkAndSetVRFFromNetwork(client *clients.AviClient) bool {
 		utils.AviLog.Infof("Using global VRF for NodePort mode")
 		return true
 	}
+	if lib.GetCloudType() == lib.CLOUD_NSXT && lib.GetServiceType() == "ClusterIP" && lib.GetCNIPlugin() != lib.NCP_CNI {
+		// Here we need to determine the right VRF for this T1LR
+		// The logic is: Get all the VRF context objects from the controller, figure out the VRF that matches the T1LR
+		// Current pagination size is set to 100, this may have to increased if we have more than 100 T1 routers.
+		uri := "/api/vrfcontext?" + "&include_name=true&cloud_ref.name=" + utils.CloudName + "&page_size=100"
+		result, err := lib.AviGetCollectionRaw(client, uri)
+		if err != nil {
+			utils.AviLog.Warnf("Get uri %v returned err %v", uri, err)
+			return false
+		}
+		elems := make([]json.RawMessage, result.Count)
+		err = json.Unmarshal(result.Results, &elems)
+		if err != nil {
+			utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+			return false
+		}
+		var foundVrf bool
+		for i := 0; i < result.Count; i++ {
+			vrf := models.VrfContext{}
+			err = json.Unmarshal(elems[i], &vrf)
+			if err != nil {
+				utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+				continue
+			}
 
-	vrfRef := *network.VrfContextRef
-	vrfName := strings.Split(vrfRef, "#")[1]
-	utils.AviLog.Infof("Setting VRF %s found from network %s", vrfName, networkName)
-	lib.SetVrf(vrfName)
+			vrfName := *vrf.Name
+			if vrf.Attrs != nil {
+				for _, v := range vrf.Attrs {
+					if *v.Key == "tier1path" && *v.Value == lib.GetT1LRPath() {
+						lib.SetVrf(vrfName)
+						utils.AviLog.Infof("Setting VRF %s found that matches the T1Lr %s", vrfName, lib.GetT1LRPath())
+						foundVrf = true
+						break
```

Comment:
should we exit from outer loop once we get first match? or we are planning to iterate through all items?

Reply:
Makes sense, I will make the change.